[bitnami/redis] Enables Redis to utilize external-dns (#8570)
bitnami/charts: bitnami/redis/README.md

To modify the application version used in this chart, specify a different version of the image using the `image.tag` parameter and/or a different repository using the `image.repository` parameter. Refer to the [chart documentation for more information on these parameters and how to use them with images from a private registry](https://docs.bitnami.com/kubernetes/infrastructure/redis/configuration/change-image-version/).

### External DNS

> This chart is equipped to allow leveraging the [`external-dns`] project.
> Doing so will enable ExternalDNS to publish the FQDN for each instance,
> in the format of `<pod-name>.<release-name>.<dns-suffix>`.
> Example, when using the following configuration:

Reviewer: ```suggestion
This chart is equipped to allow leveraging the ExternalDNS project. Doing so will enable ExternalDNS to publish the FQDN for each instance, in the format of `<pod-name>.<release-name>.<dns-suffix>`.
Example, when using the following configuration:
```

```yaml
useExternalDNS:
  enabled: true
  suffix: prod.example.org
  additionalAnnotations:
    ttl: 10
```

> On a cluster where the name of the Helm release is `a`, the hostname of a Pod
> is generated as: `a-redis-node-0.a-redis.prod.example.org`.  The IP of
> that FQDN will match that of the associated Pod.  This modifies the following
> parameters of the Redis/Sentinel configuration using this new FQDN:

Reviewer: ```suggestion
On a cluster where the name of the Helm release is `a`, the hostname of a Pod is generated as: `a-redis-node-0.a-redis.prod.example.org`. The IP of that FQDN will match that of the associated Pod. This modifies the following parameters of the Redis/Sentinel configuration using this new FQDN:
```

> On a cluster where the name of the installation is `a`, the hostname of a Pod
> is configured like so: `a-redis-node-0.a-redis.prod.example.org`.  The IP of
> that FQDN will match that of the associated Pod.  This modifies the following
> parameters of the Redis/Sentinel configuration using this new FQDN:

Reviewer 2: ```suggestion
On a cluster where the name of the Helm release is `a`, the hostname of a Pod
is generated as: `a-redis-node-0.a-redis.prod.example.org`.  The IP of
that FQDN will match that of the associated Pod.  This modifies the following
parameters of the Redis/Sentinel configuration using this new FQDN:
```

Tweaking to Helm nomenclature, and slightly tweaking the language.

* `replica-announce-ip`
* `known-sentinel`
* `known-replica`
* `announce-ip`

:warning: This requires a working installation of [`external-dns`] to be fully

> functional. :warning:

Reviewer: ``` suggestion
:warning: This requires a working installation of [`external-dns`] to be fully functional. :warning:
```

> See the documentation of [`external-dns`] for additional configuration options.
> 
> [`external-dns`]: https://github.com/kubernetes-sigs/external-dns

Reviewer: ```suggestion
See the [official ExternalDNS documentation](https://github.com/kubernetes-sigs/external-dns) for additional configuration options.
```
